staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -139,3 +140,84 @@ type CustomResourceDefinitionList struct {
 	// Items individual CustomResourceDefinitions
 	Items []CustomResourceDefinition
 }
+
+// CustomResourceValidation is a list of validation methods for CustomResources.
+type CustomResourceValidation struct {
+	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
+	OpenAPIV3Schema *JSONSchemaProps
```

Comment:
I know we can make it work and it is hard to throw away code/reviews. But other than cost of coding, there would be const of maintenance and bugs etc. Adding it as RawExtension would only add a unmarshall call to this code and get rid of many types/conversions/issues (solved and unsolved). I personally vote for simpler code in this case. We don't need to know the content of validation schema and we won't change it (unless we change the whole version of it). That is an standard spec.